test/core/util/port_server_client.cc

```diff
@@ -68,9 +68,9 @@ void grpc_free_port_using_server(int port) {
 
   grpc_init();
 
-  memset(&pr, 0, sizeof(pr));
+  pr = {};
   memset(&req, 0, sizeof(req));
```

Comment:
shouldn't the same be done here?

Reply:
I had considered it.

grpc_httpcli_request is a "trivial" type, so it's safe to memset it; so it's not needed to change it to avoid undefined/unsafe behavior or to remove a warning.

It could also be changed to use value initialization, but the grpc_http_request type does not define default member initializer values specified. This means that it would be aggregate-initialized, instead of value-initialized. In this case, it would result in the members being correctly value-initialized, though.

Since grpc_httpcli_request doesn't explicitly specify what a "default" structure should like like via default member initializer values, at least the rationale in my commit description doesn't directly apply (if it specifies what a default struct should look like, don't override that by deciding yourself it should be zeroes). Changing grpc_httpcli_request to specify default member initializer values would mean touching core/ instead of just test/, so I shied away from it to keep this more obviously not harmful.

I'd be happy to do one of the following:
a) use = {} for the three remaining memsets (they'd be aggregate initialized, which does properly value initialize the members to "0" in this case).
b) Change grpc_httpcli_request to specify default member initialization values (and also do a). But, again, I'd be touching core/ - your call.
c) (of course, leave as-is)

Just let me know your preference. :-)

Reply:
Thanks for the explanation! It's fine to leave it as-is in this case